wordpress-mobile/WordPress-Android
# Prepublishing Nudges : Publish scheduling

Fixes #11607

## Solution
1. `PublishSettingsFragment` & `PublishSettingsViewModel` from #11727 were used to create `PrepublishingPublishSettingsFragment` & `PrepublishingPublishSettingsViewModel`. 

2. The Date/Time picker & notification dialogs were using a reference to the `EditPostPublishSettingsViewModel` to update the selected dates. This was refactored to select the right `ViewModel` type depending on the screen that is using the functionality. I was going to use some form of `.class::java` approach, by passing class types from the calling fragment but decided not to and opted for enums instead 933be4a9646dad1327207e0007c5495ac1fbf948 2d790d7

## Testing
1. To test simply create a new post. 
2. Press publish. 
3. Once the bottom sheet is shown click the publish scheduling button, and you will see the `Publish` screen with the date picking mechanisms. 
4. Change the date and then go back to the home of the bottom sheet to see the changes.
5.Go to Post Settings.
6. Click on Publish.
7. Ensure you see the values that were set in the bottom sheet.

Publish immediately | Scheduled | Home screen 
--------|------- |-------
  <img src="https://user-images.githubusercontent.com/1509205/79900800-6b9d5580-83d4-11ea-864a-8a00d2ce8529.png" width="320"> | <img src="https://user-images.githubusercontent.com/1509205/79900801-6cce8280-83d4-11ea-9130-cb1f88362a6c.png" width="320"> |   <img src="https://user-images.githubusercontent.com/1509205/79900805-6d671900-83d4-11ea-94bf-0ea2e5ce18d4.png" width="320">

------------
Since the `EditPostPublishSettings` viewmodel & fragment logic was modified in this PR, it would be good to a test to ensure the functionality is still intact. 

1. Create a post.
2. Go to Post Settings.
3. Click on Publish.
4. Change the Time & Date.
5. Also, change the Notification.
6. Go back and ensure the values have been set and reflected in the Post Settings UI.

## Reviewing 

1. Only 1 reviewer is needed but anyone can review.
2. Ensure that #11727 is merged.
3. Set this base branch to feature/prepublishing_nudges
4. Remove Not Ready for Merge
5. Merge :)

## Submitter Checklist
